src/slack/slack_types.bal

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/oauth2;
+import ballerina/http;
+
+# Contains information about the channel topic.
+# 
+# + value - Value of the topic
+# + creator - The user Id of the member that created the topic
+public type Topic record {
+    string value;
+    string creator;
+};
+
+# Contains information about the channel purpose.
+# 
+# + value - Value of the purpose
+# + creator - The user Id of the member that created the channel
+public type Purpose record {
+    string value;
+    string creator;
+};
+
+# Contains information about the channel.
+# 
+# + id - Id of the channel
+# + name - Name of the channel
+# + is_channel - `true` if a channel, else `false`
+# + is_group - `true` if a group conversation, else `false`
+# + is_archived - `true` if the channel is archived
+# + is_general - `true` if the channel is the "general" channel that includes all regular members
+# + name_normalized - Normalized name
+# + parent_conversation - Name of the parent conversation if there is one
+# + creator - The user Id of the member that created the channel
+# + is_member - `true` if the calling member is part of the channel
+# + is_private - `true` if a private channel
+# + topic - Contains information about the channel topic
+# + purpose - Contains information about the channel purpose
+public type Channel record {
+    string id;
+    string name;
+    boolean is_channel;
```

Comment:
Is there a reason to use `_` in the variable names?

Reply:
Slack API's response returns these parameters and I think it is better to keep them as it is. And also we are using constructFrom to convert the json response into ballerina record. In that case also we need to keep the names as it is.